File: leetcode-contest/weekly-407/1.py
class Solution:
    def minChanges(self, n: int, k: int) -> int:
        res = 0
        i = 0
        while n or k:
            if n & 1 and not k & 1:
                res += 1
            elif k & 1 and not n & 1:
                return -1
            n = n >> 1
            k = k >> 1

        return res

File: leetcode-contest/weekly-427/1.cpp
class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> res(n);
        for (int i = 0; i < n; ++i)
        {
            int cand = i + nums[i];
            int idx = ((cand % n) + n) % n;
            res[i] = nums[idx];
        }
        return res;
    }
};